### ISSUE TITLE
Investigate `atomic_sync_purchases` job failures for non-atomic sites

### ISSUE THREAD

**Issue by gcsecsey** [Automattic/wp-calypso#73384]
Related thread: p7DVsv-fHs-p2#comment-44456

The `atomic_sync_purchases` job is sometimes failing with the "Site is not atomic" error message:
- p1675384976559749-slack-C02Q5922VGC
- p1675126236297989-slack-C02Q5922VGC

From the code it seems that this job is started with 5 second delay only if it's not an atomic site:
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Snqzva%2Qcyhtvaf%2Sngbzvp.cuc%3Se%3Qpp07op25%23410-og

Still, it seems to be started sometimes for non-atomic sites. We need to investigate the root cause and add an atomic check to prevent these job failures.

**Comment by gcsecsey** [comment]
There have been additional reports to the original P2 of the notice issue since the creation of this ticket: p7DVsv-fHs-p2#comment-44481

I've checked the last couple of reports, but there is no corresponding `atomic_sync_purchases` job failure to these, so the two issues seem to be unrelated.

**Comment by gcsecsey** [comment]
Looking back at the job error notifications, it seems that the sync issues that were caused were fixed by https://github.com/Automattic/dotcom-forge/issues/1432

**Comment by gcsecsey** [comment]
I've also tested the below scenarios to check when the job is run:

| Event | `atomic_sync_purchases` job started |
|--------|--------|
| sync purchases for "atomic without upgrade" site on SA | N |
| plan purchased on free site | N |
| plugin purchased on atomic site | Y |
| plugin purchase removed from site | Y |
| plan renewal cancelled | Y |
| plan removed from site | N | 

This is how the job is expected to run and there are no errors produced in the alerts channel. I think we can close this issue as non-reproducible.

**Comment by danielbachhuber** [comment]
Should we keep this issue open until the root cause is identified? We still need to fix the issue…

**Comment by gcsecsey** [comment]
> Should we keep this issue open until the root cause is identified? We still need to fix the issue…

@danielbachhuber Sorry, for some reason my reply in P2 got filtered, I approved it now: p7DVsv-fHs-p2#comment-45049

The more recent reports seem to be caused by another issue, not by the site not being atomic. I've tested the [scenarios I could think of](https://github.com/Automattic/wp-calypso/issues/73384#issuecomment-1476463504) regarding atomic transfer, and the job seems to be run correctly.

Regarding the issue from the latest reports on the P2 with the wrong expiry dates, is this something that @Automattic/yolo could look into?
